auth/credentials.py

```diff
@@ -83,10 +120,90 @@ def validate_credentials(auth_username, auth_password_or_token):
         logger.debug("Found credentials header for robot %s", auth_username)
         try:
             robot = model.user.verify_robot(auth_username, auth_password_or_token)
+
+            robot_user, robot_name = parse_robot_username(auth_username)
+
+            owner = model.user.get_nonrobot_user(robot_user)
```

Comment:
The same calls are already being made when verifying the username/pass. 
And these will essentially result in significantly more DB calls.

Reply:
Refactored the implementation to be based on specific exception subclasses. Except in one case I now also avoid an additional DB lookup when logging the audit event about the auth failure.